core/server/api/v2/slugs.js

```diff
@@ -0,0 +1,39 @@
+const models = require('../../models');
+const common = require('../../lib/common');
+
+const allowedTypes = {
+    post: models.Post,
+    tag: models.Tag,
+    user: models.User,
+    app: models.App
+};
+
+module.exports = {
+    docName: 'slugs',
+    generate: {
+        options: [
+            'include',
+            'type'
+        ],
+        data: [
+            'name'
+        ],
+        permissions: true,
+        validation: {
+            options: {
+                allowedTypes: allowedTypes
```

Comment:
🤔 What's the reason to use the key as `type` only? I was thinking that `allowedTypes` name kind of denotes what the object is. Also it could have been just an array with allowed types, but I tried to reuse the object we create already for this.

Reply:
The option is called `type`. How will the framework know what "allowedTypes" is? 🤔 

`type: allowedTypes`

Reply:
Solved: We use validation error for slug types directly from shared validator which checks for unknown types. 
```
validation: 
  options:
    types: allowedTypes
```